common/src/main/java/org/mcaccess/minecraftaccess/utils/NarrationUtils.java

```diff
@@ -523,36 +523,16 @@ private static String getSignInfo(SignBlockEntity signEntity, LocalPlayer player
     private static @NotNull Tuple<String, String> getCropsInfo(Block block, BlockState blockState, String toSpeak, String currentQuery) {
         int currentAge, maxAge;
 
-        switch (block) {
-            case CropBlock ignored -> {
-                if (block instanceof BeetrootBlock) {
-                    // Beetroot have a different max_age of 3
-                    currentAge = blockState.getValue(BeetrootBlock.AGE);
-                    maxAge = BeetrootBlock.MAX_AGE;
-                } else if (block instanceof TorchflowerCropBlock) {
-                    currentAge = blockState.getValue(TorchflowerCropBlock.AGE);
-                    maxAge = TorchflowerCropBlock.MAX_AGE;
-                } else {
-                    // While wheat, carrots, and potatoes have max_age of 7
-                    currentAge = blockState.getValue(CropBlock.AGE);
-                    maxAge = CropBlock.MAX_AGE;
-                }
-            }
-            case CocoaBlock ignored -> {
-                currentAge = blockState.getValue(CocoaBlock.AGE);
-                maxAge = CocoaBlock.MAX_AGE;
-            }
-            case NetherWartBlock ignored -> {
-                currentAge = blockState.getValue(NetherWartBlock.AGE);
-                maxAge = NetherWartBlock.MAX_AGE;
-            }
-            case PitcherCropBlock ignored -> {
-                currentAge = blockState.getValue(PitcherCropBlock.AGE);
-                maxAge = PitcherCropBlock.MAX_AGE;
-            }
-            case null, default -> {
-                return new Tuple<>(toSpeak, currentQuery);
-            }
+        if (block instanceof CropBlock) {
+            CropBlock crop = (CropBlock)block;
```

Comment:
This should use a binding in the instanceof expression instead of a cast. Same below.
```suggestion
        if (block instanceof CropBlock crop) {
```